specs/language/resources.tex

```diff
@@ -0,0 +1,755 @@
+
+\begin{HLSL}
+  bool CheckAccessFullyMapped(in uint status);
+\end{HLSL}
+
+This function returns true is the value was accessed from a fully committed resource,
+or from fully mapped pages of a sparse resource.
+If any part of the return value was from unmapped memory, false is returned.
+
+\Sec{Resource Binding}{Resources.binding}
+
+Resources are bound to external memory using virtual registers within logical registers spaces.
+These can be specified explicitly using attributes after the global resource declaration
+or implicitly by leaving them off.
+
+\begin{grammar}
+  \define{resource-binding}\br
+  \terminal{: register(} register-type bind-number \terminal{)}\br
+  \terminal{: register(} register-type bind-number , \terminal{space} bind-number \terminal{)}\br
+  \define{register-type} \textnormal{one of}\br
+  \terminal{t u b s}\br
+  \define{bind-number}\br
+  digit\br
+  bind-number digit\br
+\end{grammar}
+
+The register type indicates whether the binding is for a read-only (\texttt{t}), a writable (\texttt{u}),
+a constant buffer (\texttt{b}), or a sampler (\texttt{s}).
+The register bind number indicates the register number at which the resource variable begins.
+The optional second parameter indicates the virtual register space that the register belongs to.
+
+Sample syntax of virtual register binding attributes:
+\begin{HLSL}
+  Buffer<float> robuf : register(t0, space0);
+  RWBuffer<float> rwbuf : register(u0, space0);
+  SamplerState samp : register(s0, space0);
+  cbuffer cbuf : register(b0, space0) { float f; };
+\end{HLSL}
+
+Each resource must have a unique register. Different resources cannot occupy the same register,
+but read-only and writable buffers have separate namespaces and different logical register spaces have independent
+sets of register numbers. Meaning the following is all allowed:
+\begin{HLSL}
+  Buffer<float> mybuf : register(t0, space0);
+  Buffer<float> yourbuf : register(t0, space1);
+  RWBuffer<float> hisbuf : register(u0, space0);
+  RWBuffer<float> herbuf : register(u0, space1);
+\end{HLSL}
+
+A aggregate resource type can be:
+\begin{itemize}
+\item a struct containing one or more resources types
+\item an array of resource types
+\item an array or struct containing either of the above
+\end{itemize}
+
+When aggregate resource types have register annotations,
+they occupy the first register they specify as well as however many additional sequential registers
+the aggregate requires to assign a register to each of its elements that correspond to the given register type.
+Multiple register annotations can be placed on aggregates involving structs where each corresponds to a distinct
+register type that will apply to all the registers of that type in the aggregate.
+
+\begin{note}
+Resource types contained in structs are only allocated registers when they are explicitly used.
+This includes elements of arrays of resources as such array elements must be indexed with literals.
```

Comment:
```suggestion
This includes elements of arrays of resources, as such array elements must be indexed with literals.
```

Reply:
I also don't think this comma is needed. "As" serves the same purpose as "since", which I will replace it with for similar reasons as above. 

This says that if it is a subordinate clause following the main close, no comma is needed. It is needed when it is a subordinate clause that precedes the main clause e.g. "Since such array elements must be indexed with literals, this includes elements of arrays of resources."